src/components/organisms/UiMenu/UiMenu.vue

```diff
@@ -62,125 +50,170 @@ export interface MenuProps {
+        await focusElement(nextElement.value.itemTemplateRefs.content.$el, true);
+        initialElement.value.tabindex = -1;
       }
       break;
     case 'Home':
     case 'PageUp':
-      if (firstMenuItem.value && focusedMenuItem.value) {
-        focusedMenuItem.value.tabindex = -1;
-        firstMenuItem.value.tabindex = 0;
-        focusElement(firstMenuItem.value.$el.querySelector('.ui-button'));
+      if (firstElement.value
+          && initialElement.value) {
+        await focusElement(firstElement.value.itemTemplateRefs.content.$el, true);
+        initialElement.value.tabindex = -1;
       }
       break;
     case 'End':
     case 'PageDown':
-      if (lastMenuItem.value && focusedMenuItem.value) {
-        focusedMenuItem.value.tabindex = -1;
-        lastMenuItem.value.tabindex = 0;
-        focusElement(lastMenuItem.value.$el.querySelector('.ui-button'));
+      if (lastElement.value
+          && initialElement.value) {
+        await focusElement(lastElement.value.itemTemplateRefs.content.$el, true);
+        initialElement.value.tabindex = -1;
       }
       break;
     case 'Tab':
-      const lastFocusedElement = ref(focusedMenuItem.value);
-      setTimeout(() => {
-        if(!lastFocusedElement.value) return;
-        lastFocusedElement.value.tabindex = -1;
-        if (selectedMenuItem.value) {
-          selectedMenuItem.value.tabindex = 0;
-          return;
-        }
-        if (firstMenuItem.value) {
-          firstMenuItem.value.tabindex = 0;
-        }
-      }, 0);
+      if (initialElement.value === activeElement) {
+        return;
+      }
+      if (activeElement) {
+        activeElement.tabindex = 0;
+        // We need to use setTimeout() because nextTick() does not bring the desired effect.
+        window.setTimeout(() => {
+          activeElement.tabindex = -1;
+        }, 0);
+      }
+      break;
+    default:
       break;
-    default: break;
   }
 };
-defineExpose({
-  menuItems,
-  firstMenuItem,
-  selectedMenuItem,
-});
-export interface MenuEmits {
-  (e: 'itemsLoaded'): void;
-}
-const emit = defineEmits<MenuEmits>();
-const hasMenuItems = computed(() => (
-  menuItems.value.length > 0
-));
-const setItemsNotReachable = () => {
-  [ ...menuItems.value ].forEach((item) => {
-    item.tabindex = item.$el.querySelector('.ui-button') ? -1 : 0;
-  });
-  if (selectedMenuItem.value) {
-    selectedMenuItem.value.tabindex = 0;
-  } else if(firstMenuItem.value) {
-    firstMenuItem.value.tabindex = 0;
+const isFocused = ref(false);
+const scrollLock = (event: KeyboardEvent) => {
+  const { key } = event;
+  switch (key) {
+    case 'ArrowUp':
+    case 'ArrowDown':
+    case 'Home':
+    case 'PageUp':
+    case 'End':
+    case 'PageDown':
+      event.preventDefault();
+      break;
+    default:
+      break;
   }
-  emit('itemsLoaded');
 };
-watch(hasMenuItems, async (hasItems) => {
-  if (hasItems) {
-    await nextTick();
-    setItemsNotReachable();
+watch(isFocused, (value) => {
```

Comment:
FEAT: prevent moving paga by arrows when the user uses arrows to navigate on UiMenu.